salt/minion.py

```diff
@@ -3205,6 +3205,28 @@ def _post_master_init(self, master):
             self.schedule.delete_job(master_event(type='alive', master=self.opts['master']), persist=True)
             self.schedule.delete_job(master_event(type='failback'), persist=True)
 
+        # proxy keepalive
+        proxy_alive_fn = fq_proxyname+'.alive'
+        if proxy_alive_fn in self.proxy and 'status.proxy_reconnect' in self.functions and \
+           ('proxy_keep_alive' not in self.opts or ('proxy_keep_alive' in self.opts and self.opts['proxy_keep_alive'])):
```

Comment:
BTW, since we just re-opened this, then I would write it a bit more readable:
```python
        if (proxy_alive_fn in self.proxy
            and 'status.proxy_reconnect' in self.functions 
            and 'proxy_keep_alive' not in self.opts or 'proxy_keep_alive' in self.opts
            and self.opts['proxy_keep_alive']):
```